modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/raft/snapshot/outgoing/OutgoingSnapshot.java

```diff
@@ -512,13 +514,20 @@ public boolean addRowIdToSkip(RowId rowId) {
     public boolean alreadyPassed(int tableId, RowId rowId) {
         assert mvOperationsLock.isLocked() : "MV operations lock must be acquired!";
 
+        if (mvPartitionDeliveryState == null) {
+            // We haven't started sending MV data yet.
+            return false;
+        }
+
         if (finishedMvData()) {
             return true;
         }
 
-        return partitionDeliveryState != null
-                && tableId <= partitionDeliveryState.currentTableId()
-                && rowId.compareTo(partitionDeliveryState.currentRowId()) <= 0;
+        if (tableId == mvPartitionDeliveryState.currentTableId()) {
+            return rowId.compareTo(mvPartitionDeliveryState.currentRowId()) <= 0;
```

Comment:
Could you please add a comment saying that `currentRowId()` is already sent? Previous version of the code operated with `lastRowId` variable that made this obvious, but now it is not that obvious anymore